cmake/templates/BoostConfig.cmake.in

```diff
@@ -51,10 +50,16 @@ if(NOT TARGET "Boost::boost")
       INTERFACE_INCLUDE_DIRECTORIES "${_root_path}/include"
   )
   if(MSVC)
-    set_target_properties(
+    #stop MSVC from trying to auto-link libs
+	set(compile_definitions "BOOST_ALL_NO_LIB=1")
+    if(@BOOST_ALL_DYN_LINK@) #required for MSVC when using shared libs
+	  set(compile_definitions "${compile_definitions} BOOST_ALL_DYN_LINK")
```

Comment:
Setting INTERFACE_COMPILE_DEFINITIONS using a target property will not take a list. I don't know if this is a hack but sending it this way seems to be accepted. Using target_compile_definitions would probably be better I just wasn't sure about syntax as I am not that familiar with it (or most of this level of cmake for that matter), I am assuming it could be done as follows

```
target_compile_definitions(Boost::boost PRIVATE "BOOST_ALL_NO_LIB=1")
if(@BOOST_ALL_DYN_LINK@)
  target_compile_definitions(Boost::boost PRIVATE "BOOST_ALL_DYN_LINK")
```